DOC: Move all docs into en/ subdirectory

d0sboots opened the pull request:
PR #1502 is working on adding a Chinese translation to the docs. In general, I encouraged this (in #1452) as a path towards getting useful translated content in the game without requiring a massive refactor/rearchitecting of everything.

To support this, this takes the first step of moving our docs into an en/ subdirectory, so that other languages can live alongside. No effort is made at this time to support or select between alternate languages; this is a pure-rename refactor.

Testing: In-game docs still work (for dev server). Ctrl-click is temporarily broken, but will work once this PR is submitted (points at the new location).

d0sboots committed:
DOC: Move all docs into en/ subdirectory

PR #1502 is working on adding a Chinese translation to the docs. In
general, I encouraged this (in #1452) as a path towards getting useful
translated content in the game without requiring a massive
refactor/rearchitecting of everything.

To support this, this takes the first step of moving our docs into an
en/ subdirectory, so that other languages can live alongside. No effort
is made at this time to support or select between alternate languages;
this is a pure-rename refactor.

d0sboots commented:
@Snarling for signoff on design/overall goal

catloversg commented:
This change looks good to me. After merging it, other contributors can start translating documentation.

d0sboots commented:
Reminder to myself to fix/redo this PR and submit it

d0sboots committed:
Do the actual rename

d0sboots committed:
prettier

d0sboots commented:
I have updated this PR, and am submitting it now, because it touches all documentation.

I changed it slightly - it does actually look at the user's locale now (is no longer hardcoded), but there is a fallback. I tested with "ja" as the locale and it seems to work fine. We can patch it after if there is an issue; I don't want to block the whole thing since the docs refactor is big.
